### PR TITLE
Switch to using PublishedMetada.create_from_file()

### PR DESCRIPTION
**Pull request by dkliban**
Required PR: pulp/pulpcore#303

re: #5304
https://pulp.plan.io/issues/5304

### CODE REVIEW

File: CHANGES/5304.misc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Required PR: https://github.com/pulp/pulpcore/pull/303/
+
+re: #5304
+https://pulp.plan.io/issues/5304
```
Comment (mdellweg) [inline CHANGES/5304.misc]: this sounds like the information to go in the commit message.

File: pulp_deb/app/tasks/publishing.py [hunk 2]
```diff
@@ -111,27 +111,21 @@ def publish(repository_version_pk, simple=False, structured=False):
                     _add_to_release(release, package_index_path)
                     _add_to_release(release, gz_package_index_path)
 
-                    package_index = PublishedMetadata(
-                        relative_path=package_index_path,
-                        publication=publication,
-                        file=File(open(package_index_path, "rb")),
+                    package_index = PublishedMetadata.create_from_file(
+                        publication=publication, file=File(open(package_index_path, "rb"))
```
Comment (mdellweg) [inline pulp_deb/app/tasks/publishing.py]: How is the relative_path determined in this PublishedMetadata?

Comment (dkliban) [reply pulp_deb/app/tasks/publishing.py]: If the relative_path is passed in to create_from_file, the that value is used. If it is not passed in, the file name from the file object is used. 

https://docs.pulpproject.org/en/pulpcore-plugin/nightly/api-reference/models.html#pulpcore.plugin.models.PublishedMetadata.create_from_file

File: CHANGES/5304.misc (hunk 1 above)
Comment (dkliban) [reply CHANGES/5304.misc]: yep ... fixed it now.

**Pull request merged**